### PR TITLE
fix(e2e): stabilize v2 executor node types save test (AAP-89592)

### PR DESCRIPTION
**Pull request by goneri**
Avoid semantic-zoom getByText races on approval branches, ignore POST /workflows/validate when capturing the save payload, and wait for a real saved URL before listing the workflow.

Co-authored-by: Cursor <cursoragent@cursor.com>

### CODE REVIEW

**Comment by goneri** [comment]
PR Rebased on top of https://github.com/syntara-orchestration/syntara/pull/533

File: frontend/packages/syntara-ui/e2e/v2-workflow-migration.spec.ts [hunk 1]
```diff
@@ -383,6 +389,8 @@ test.describe('V2 Workflow Schema Migration', () => {
     await selectProjectIfRequired(app)
     await app.getByPlaceholder('Workflow name').fill(workflowName)
     await clickSaveAndWait(app)
+    await app.getByRole('button', { name: 'Save' }).click()
```
Comment (gitdallas) [inline frontend/packages/syntara-ui/e2e/v2-workflow-migration.spec.ts]: nit, it looks like you are doing a click on "save" right after `clickSaveAndWait`

**Review by gitdallas — APPROVED** [review]
lgtm, one nit

File: frontend/packages/syntara-ui/e2e/v2-workflow-migration.spec.ts [hunk 2]
```diff
@@ -383,6 +389,8 @@ test.describe('V2 Workflow Schema Migration', () => {
...
+    await expect(app).toHaveURL(/workflow-builder\/(?!new\b).+/, { timeout: 15_000 })
```
Comment (Rishi625) [inline frontend/packages/syntara-ui/e2e/v2-workflow-migration.spec.ts]: `clickSaveAndWait` (helpers/workflow-save.ts) already guarantees the save response, the settle (`Save` visible again), and — since this page starts at `/workflow-builder/new` — the `(?!new)` navigation. So this extra bare `Save` click is redundant, and risky: once the save settles, the button is `isAriaDisabled` (clean form), and with no `actionTimeout` in `playwright.config.ts` a click landing after settle waits out the timeout instead of failing fast (see the `SAVE_CLICK_TIMEOUT` comment in workflow-save.ts). If it lands before settle it's a no-op redundant PATCH. Either way it can only add flake to the test this PR is stabilizing — and it can't create a duplicate workflow, so no product risk. Suggest deleting the click and keeping just the 15s URL expect (if the 10s settle inside `clickSaveAndWait` proved short).

File: frontend/packages/syntara-ui/e2e/v2-workflow-migration.spec.ts [hunk 3]
```diff
@@ -81,6 +81,13 @@ function getV2DefFromRequest(request: { postDataJSON: () => unknown }): V2Workfl
   return payload.workflow_definition
 }
 
+/** POST /workflows create only — excludes /validate and /{id}/… sub-resources. */
+function isWorkflowCreateRequest(request: { url: () => string; method: () => string }): boolean {
+  if (request.method() !== 'POST') return false
+  const path = new URL(request.url()).pathname.replace(/\/$/, '')
+  return path.endsWith('/workflows')
+}
```
Comment (Rishi625) [inline frontend/packages/syntara-ui/e2e/v2-workflow-migration.spec.ts:89]: Nit: this duplicates the `isWorkflowSaveResponse` concept in `helpers/workflow-save.ts` (which documents the same `/validate` trap). Keeping request interception is the right call here — for a schema-migration test the *sent* payload is the artifact under test — but the matcher would fit better next to `isWorkflowSaveResponse` in `workflow-save.ts` than inline in the spec. Non-blocking.

**Review by Rishi625 — COMMENTED** [review]
Only ask is the redundant Save click; otherwise LGTM from the backend side.

**Review by acosferreira — APPROVED** [review]

**Review by Rishi625 — APPROVED** [review]
